### ISSUE TITLE
Refactor Mode class methods to static and remove WC_Payments::mode()

### ISSUE THREAD

**Issue by gpressutto5** [Automattic/woocommerce-payments#8039]
### Description
After #8003, the Mode class does not have a constructor anymore, so an instance is not needed. We are still saving it to [WC_Payments](https://github.com/Automattic/woocommerce-payments/blob/f63d75f2c2a83e8dce804fa0b248e86eefbeeeac/includes/class-wc-payments.php#L296) and using [WC_Payments::mode()](https://github.com/Automattic/woocommerce-payments/blob/f63d75f2c2a83e8dce804fa0b248e86eefbeeeac/includes/class-wc-payments.php#L662) to use it. This makes it so we can only use it after it has been [initialized](https://github.com/Automattic/woocommerce-payments/blob/f63d75f2c2a83e8dce804fa0b248e86eefbeeeac/includes/class-wc-payments.php#L296C19-L296C19).
We can refactor its methods to be static and then update all usages of `WC_Payments::mode()` to use `Mode` directly.

### Acceptance criteria

* `WC_Payments::mode()` is removed

**Comment by RadoslavGeorgiev** [comment]
Hey @gpressutto5, there are some unannounced efforts going on within the `src` directory, as a part of the Payment Processing Re-Engineering Project (paJDYF-9hL-p2). Even though the project is on hold, there are some principles in there that probably should still be followed, namely this (found [here](https://github.com/Automattic/woocommerce-payments/blob/develop/src/README.md?plain=1#L257)):

> `src` classes should only have pure static methods.

The point here is that unless a method is pure, it should not be static because that will require workarounds when testing. In this case, I'd say that the same applies to the whole class: if there is a constructor, and it is not pure, likely the whole class should focus on being static.

We didn't get to promote this strategy, as the project got put on hold before we had the opportunity.

---

Here are things I have on my mind:

1. It will make testing worse. There are classes within `src` (that has 100% test coverage) that rely on `Mode`, and those will be degraded.
2. The class will still need to be retrieved through a method before being used, as initialization is required. 
3. I'm almost certain that we'll like to introduce a class for settings in the future, accessing as a proxy to the option. That will require us to transform it back to non-static.

Overall, how is this an improvement?

**Comment by gpressutto5** [comment]
Hi @RadoslavGeorgiev, thanks for pointing that out! I can’t see anything requiring a method or initialization if we started using static methods. The greatest advantage of refactoring it to static would be using it before WC_Payments finishes initializing, also allowing us to log errors that might have prevented WC_Payments from initializing successfully.
That said, I can see how that impacts unit testing, and knowing we will have a settings class in the future also impacts this decision, so I’m closing this issue.
But I suggest we consider moving it to the Services Container so it can be injected, and we can remove one static method and property from WC_Payments.

**Comment by RadoslavGeorgiev** [comment]
>  I can’t see anything requiring a method or initialization if we started using static methods.

I thought `maybe_init` _was_ called by the constructor, but now I see that it's called within every method. I think the reason for that was the circular dependency between the gateway and the `Mode` class. Now that the dependency on the gateway is gone, it would make more sense to do it in the constructor.

> Also allowing us to log errors that might have prevented `WC_Payments` from initializing successfully

We discussed using the logger (also dependent on the mode) before initializing `WC_Payments` here last week: p1705680758641529/1705652438.299549-slack-CGGCLBN58. The discussion might be useful.

>  But I suggest we consider moving it to the Services Container so it can be injected, and we can remove one static method and property from `WC_Payments`.

That's a great idea, now that `Mode` does not depend on the gateway. I'm not sure opening an issue about that would make sense now, but I'll definitely keep in in mind once we resurect the RPP project.
